### PR TITLE
TerminalService

### PR DESCRIPTION
**Pull request by nagulmeeras**

### CODE REVIEW

File: src/main/scala/com/imaginea/Main.scala [hunk 1]
```diff
@@ -1332,9 +1351,13 @@ object Main extends App {
   }
 
   def getKeyById(userId: Long, keyId: Long): Option[KeyPairInfo] = {
-    User.fromNeo4jGraph(userId) match {
-      case Some(user) => user.publicKeys.dropWhile(_.id.get != keyId).headOption
-      case None => None
+    User.fromNeo4jGraph(userId).flatMap { user =>
+      user.publicKeys.find { keyPair =>
+        keyPair.id match {
```
Comment (eklavya) [inline src/main/scala/com/imaginea/Main.scala]: use contains

Comment (nagulmeeras) [reply src/main/scala/com/imaginea/Main.scala]: Changed it to contains

File: src/main/scala/com/imaginea/activegrid/core/models/ListCommand.scala [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  var list: Boolean = false
+
+  @Parameter(description = "the current context to execute command on")
+  var newContext: java.util.List[String] = new java.util.ArrayList()
+
+  val logger = Logger(LoggerFactory.getLogger(getClass.getName))
+
+  override def execute(commandExecutionContext: CommandExecutionContext, lines: List[Line]): CommandResult = {
+    val targetContext = newContext.toList match {
+      case contexts if contexts.size == 1 =>
```
Comment (eklavya) [inline src/main/scala/com/imaginea/activegrid/core/models/ListCommand.scala]: a simple if else would be much more readable

Comment (nagulmeeras) [reply src/main/scala/com/imaginea/activegrid/core/models/ListCommand.scala]: Changed it now

**Pull request merged**